File: backend/internal/domain/models/electricity_reading.go
package models

type ElectricityReading struct {
	ID                 uint   `gorm:"primaryKey" json:"id"`
	ElectricityMeterID uint   `json:"electricity_meter_id"`
	LowerReading       string `json:"lower_reading"`
	UpperReading       string `json:"upper_reading"`
	ReadingDate        string `json:"reading_date"`
}

File: frontend/src/app/models/electricity_reading.ts
export interface ElectricityReading {
    id: number;
    electricity_meter_id: number;
    lower_reading: string;
    upper_reading: string;
    reading_date: string;
};  

export interface CreateElectricityReading {
    electricity_meter_id: number;
    lower_reading: string;
    upper_reading: string;
}

export interface CreateElectricityReadingWithDate extends CreateElectricityReading {
    reading_date: string;
}